src/utils/filter-utils.ts

```diff
@@ -1,16 +1,42 @@
 import isEqual from 'lodash/isEqual';
 
-import { Filter } from '@yext/search-core';
+import { FieldValueFilter, StaticFilter } from '@yext/search-core';
 
 /**
- * Returns true if the two given filters are the same.
+ * Returns true if the two given field value filters are the same.
  *
- * @param thisFilter - The first filter to compare
- * @param otherFilter - The second filter to compare
- * @returns Whether the two filters are the same or not
+ * @param thisFilter - The first field value filter to compare
+ * @param otherFilter - The second field value filter to compare
+ * @returns Whether the two field value filters are the same or not
  */
-export function areFiltersEqual(thisFilter: Filter, otherFilter: Filter): boolean {
+export function areFieldValueFiltersEqual(
+  thisFilter: FieldValueFilter,
+  otherFilter: FieldValueFilter
+): boolean {
   return thisFilter.fieldId === otherFilter.fieldId
-  && thisFilter.matcher === otherFilter.matcher
-  && isEqual(thisFilter.value, otherFilter.value);
+    && thisFilter.matcher === otherFilter.matcher
+    && isEqual(thisFilter.value, otherFilter.value);
+}
+
+/**
+ * Returns true if the two given static filters are the same.
+ *
+ * @param thisFilter - The first static filter to compare
+ * @param otherFilter - The second static filter to compare
+ * @returns Whether the two static filters are the same or not
+ */
+export function areStaticFiltersEqual(thisFilter: StaticFilter, otherFilter: StaticFilter): boolean {
+  if (thisFilter.kind === 'fieldValue') {
+    return otherFilter.kind === 'fieldValue'
+      ? areFieldValueFiltersEqual(thisFilter, otherFilter)
+      : false;
+  }
+
+  if (otherFilter.kind === 'fieldValue') {
+    return false;
+  }
+
+  return thisFilter.combinator === otherFilter.combinator
+    && thisFilter.filters.length === otherFilter.filters.length
+    && thisFilter.filters.every((f, index) => areStaticFiltersEqual(f, otherFilter.filters[index]));
```

Comment:
I suppose order matters? We wouldn't want to consider two filter arrays [A, B] and [B, A] be equal?

Reply:
I originally thought that it might be better to differentiate based on order because people may want to display the combined filters based on that. But, thinking about it more, I don't think Headless should be dictated by display concerns, and these filter arrays still technically result in the same filter. I'll update to remove the checks on the order!